app/src/main/java/org/sunsetware/phocid/data/Lyrics.kt

```diff
@@ -42,16 +47,18 @@ fun parseLrc(lrc: ByteArray, charsetName: String?): Lyrics {
 fun parseLrc(lrc: String): Lyrics {
     return Lyrics(
         lrc.lines()
-            .mapNotNull { line ->
-                lrcRegex.matchEntire(line.trimAndNormalize())?.let {
-                    val timestamp =
-                        it.groups["minutes"]!!.value.toInt().minutes +
-                            it.groups["seconds"]!!.value.toInt().seconds +
-                            (it.groups["milliseconds"]?.value?.toInt()?.milliseconds
-                                ?: (it.groups["centiseconds"]!!.value.toInt() * 10).milliseconds)
-                    val text = it.groups["text"]!!.value.trim()
-                    Pair(timestamp, text)
-                }
+            .fastFlatMap { line ->
```

Comment:
Google is famous for arbitrarily deprecating things. Don't use their API whenever possible. Also I think this is an internal API they forgot to mark private - the [docs](https://android.googlesource.com/platform/frameworks/support/+/refs/heads/androidx-main/compose/ui/ui-util/src/commonMain/kotlin/androidx/compose/ui/util/ListUtils.kt#522) mentioned "Do not use for collections that come from public APIs".

Reply:
ah, fair enough. i kind of misread that as "only use when you know that random access is fast" (and `.lines()` produces an arraylist). but i have to admit that regardless this would have been unnecessary and error prone optimization